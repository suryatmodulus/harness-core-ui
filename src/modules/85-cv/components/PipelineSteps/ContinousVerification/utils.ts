/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { getMultiTypeFromValue, MultiTypeInputType, RUNTIME_INPUT_VALUE, SelectOption } from '@wings-software/uicore'
import { isEmpty, isNull, isUndefined, omit, omitBy, set } from 'lodash-es'
import { yupToFormErrors } from 'formik'
import * as Yup from 'yup'
import type { UseStringsReturn } from 'framework/strings'
import { getDurationValidationSchema } from '@common/components/MultiTypeDuration/MultiTypeDuration'
import type { ContinousVerificationData, spec } from './types'
import {
  VerificationSensitivityOptions,
  durationOptions,
  baseLineOptions,
  trafficSplitPercentageOptions,
  SensitivityTypes
} from './constants'

/**
 * checks if a field is a runtime input.
 * @param field
 * @returns boolean
 */
export function checkIfRunTimeInput(field: string | SelectOption | number | undefined): boolean {
  return getMultiTypeFromValue(field as string) === MultiTypeInputType.RUNTIME
}

/**
 * validates a particular field for errors.
 * @param fieldValue
 * @param fieldKey
 * @param data
 * @param errors
 * @param getString
 * @param isRequired
 */
export function validateField(
  fieldValue: string,
  fieldKey: string,
  data: ContinousVerificationData,
  errors: any,
  getString: UseStringsReturn['getString'],
  isRequired = true
): void {
  if (checkIfRunTimeInput(fieldValue) && isRequired && isEmpty(data?.spec?.spec && data?.spec?.spec[fieldKey])) {
    set(errors, `spec.spec.${fieldKey}`, getString('fieldRequired', { field: fieldKey }))
  }
}

/**
 * validates timeout field.
 * @param template
 * @param getString
 * @param data
 * @param errors
 */
export function validateTimeout(
  getString: UseStringsReturn['getString'],
  data: ContinousVerificationData,
  errors: any,
  template?: ContinousVerificationData,
  isRequired = true
): void {
  if (checkIfRunTimeInput(template?.timeout)) {
    let timeout: Yup.ObjectSchema
    if (isRequired) {
      timeout = Yup.object().shape({
        timeout: getDurationValidationSchema({ minimum: '10s' }).required(getString?.('validation.timeout10SecMinimum'))
      })
    } else {
      timeout = Yup.object().shape({
        timeout: getDurationValidationSchema({ minimum: '10s' })
      })
    }

    try {
      timeout.validateSync(data)
    } catch (e) {
      if (e instanceof Yup.ValidationError) {
        const err = yupToFormErrors(e)
        Object.assign(errors, err)
      }
    }
  }
}

/**
 * returns yaml data for spec field.
 * @param specInfo
 * @returns spec
 */
export function getSpecYamlData(specInfo?: spec, type?: string): spec {
  let validspec = omitBy(specInfo, v => isUndefined(v) || isNull(v) || v === '')
  switch (type) {
    case 'LoadTest':
      validspec = omit(validspec, ['trafficsplit'])
      break
    case 'Bluegreen':
    case 'Canary':
      validspec = omit(validspec, ['baseline'])
      break
    default:
      break
  }

  Object.keys(validspec).map((key: string) => {
    //TODO logic in if block will be removed once backend api is fixed : https://harness.atlassian.net/browse/CVNG-2481
    if (key === 'sensitivity') {
      validspec[key] = validspec[key].value
        ? SensitivityTypes[validspec[key].value as keyof typeof SensitivityTypes]
        : validspec[key]
    } else {
      validspec[key] = validspec[key].value ? validspec[key].value : validspec[key]
    }
  })

  return validspec
}

/**
 * returns forms data for spec field.
 * @param specInfo
 * @returns spec
 */
export function getSpecFormData(specInfo: spec | undefined): spec {
  const validspec: spec | undefined = { ...specInfo }
  if (specInfo) {
    Object.keys(specInfo).map((key: string) => {
      switch (key) {
        case 'sensitivity':
          setFieldData(validspec, 'sensitivity', VerificationSensitivityOptions)
          break
        case 'duration':
          setFieldData(validspec, 'duration', durationOptions)
          break
        case 'baseline':
          setFieldData(validspec, 'baseline', baseLineOptions)
          break
        case 'trafficsplit':
          setFieldData(validspec, 'trafficsplit', trafficSplitPercentageOptions)
          break
        default:
      }
    })
  }
  return validspec
}

/**
 * sets particular field data
 * @param validspec
 * @param field
 * @param options
 */
export function setFieldData(validspec: spec | undefined, field: string, fieldOptions: SelectOption[]): void {
  //finding the complete option if the field is fixed input
  if (validspec && validspec[field] && validspec[field] !== RUNTIME_INPUT_VALUE) {
    //TODO logic in if block will be removed once backend api is fixed : https://harness.atlassian.net/browse/CVNG-2481
    if (field === 'sensitivity') {
      validspec[field] = fieldOptions.find(
        (el: SelectOption) =>
          SensitivityTypes[el.value as keyof typeof SensitivityTypes] === (validspec && validspec[field])
      )
    } else {
      validspec[field] = fieldOptions.find((el: SelectOption) => el.value === (validspec && validspec[field]))
    }
  }
}
