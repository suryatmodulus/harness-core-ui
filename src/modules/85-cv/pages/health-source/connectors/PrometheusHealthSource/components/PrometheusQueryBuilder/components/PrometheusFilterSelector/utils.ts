/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MultiSelectOption } from '@wings-software/uicore'

export function updateLabelOfSelectedFilter(
  selectedValue: string,
  item: MultiSelectOption,
  items: MultiSelectOption[]
): MultiSelectOption {
  const currentItem = items.find(listItem => listItem.value === item.value)
  if (currentItem) {
    const splitString = (currentItem.label as string).split(':')[0]
    return { ...currentItem, label: `${splitString}:${selectedValue}` }
  }

  return item
}

export function updateMultiSelectOption(
  newOption: MultiSelectOption,
  currentList: MultiSelectOption[]
): MultiSelectOption[] {
  if (!currentList?.length) {
    return []
  }

  const newOptionIndex = currentList.findIndex(listItem => listItem.value === newOption.value)
  if (newOptionIndex > -1) {
    currentList[newOptionIndex] = newOption
  }

  return Array.from(currentList)
}
