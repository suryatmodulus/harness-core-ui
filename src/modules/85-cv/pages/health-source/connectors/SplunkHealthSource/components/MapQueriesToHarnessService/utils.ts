/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import type { UseStringsReturn } from 'framework/strings'
import { initialFormData, MapSplunkToServiceFieldNames } from './constants'
import type { MapSplunkQueryToService } from './types'

type UpdateSelectedQueriesMap = {
  updatedMetric: string
  oldMetric: string
  mappedMetrics: Map<string, MapSplunkQueryToService>
  formikProps: FormikProps<MapSplunkQueryToService | undefined>
}

export function updateSelectedMetricsMap({
  updatedMetric,
  oldMetric,
  mappedMetrics,
  formikProps
}: UpdateSelectedQueriesMap): any {
  const updatedMap = new Map(mappedMetrics)

  // in the case where user updates query name, update the key for current value
  if (oldMetric !== formikProps.values?.metricName) {
    updatedMap.delete(oldMetric)
  }

  // if newly created query create form object
  if (!updatedMap.has(updatedMetric)) {
    updatedMap.set(updatedMetric, { ...initialFormData, metricName: updatedMetric })
  }

  // update map with current form data
  if (formikProps.values?.metricName) {
    updatedMap.set(formikProps.values.metricName, formikProps.values as MapSplunkQueryToService)
  }
  return { selectedMetric: updatedMetric, mappedMetrics: updatedMap }
}

export function validateMappings(
  getString: UseStringsReturn['getString'],
  createdMetrics: string[],
  selectedMetricIndex: number,
  values?: MapSplunkQueryToService
): { [fieldName: string]: string } {
  const requiredFieldErrors = {
    ...(!values?.metricName && {
      [MapSplunkToServiceFieldNames.METRIC_NAME]: getString('cv.monitoringSources.queryNameValidation')
    }),
    ...(!values?.query && {
      [MapSplunkToServiceFieldNames.QUERY]: getString('cv.monitoringSources.gco.manualInputQueryModal.validation.query')
    }),
    ...(!values?.serviceInstance && {
      [MapSplunkToServiceFieldNames.SERVICE_INSTANCE]: getString(
        'cv.monitoringSources.gcoLogs.validation.serviceInstance'
      )
    })
  }

  const duplicateNames = createdMetrics?.filter((metricName, index) => {
    if (index === selectedMetricIndex) {
      return false
    }
    return metricName === values?.metricName
  })

  if (values?.metricName && duplicateNames.length) {
    requiredFieldErrors[MapSplunkToServiceFieldNames.METRIC_NAME] = getString(
      'cv.monitoringSources.gcoLogs.validation.queryNameUnique'
    )
  }

  return requiredFieldErrors
}
