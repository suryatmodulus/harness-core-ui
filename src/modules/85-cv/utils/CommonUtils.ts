/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { get } from 'lodash-es'
import { Color, Utils, Views, SelectOption, MultiSelectOption } from '@wings-software/uicore'
import type { UseStringsReturn } from 'framework/strings'
import type { ResponseListEnvironmentResponse, EnvironmentResponse } from 'services/cd-ng'
import type { StringsMap } from 'stringTypes'
import type { MonitoredServiceEnum } from '@cv/pages/monitored-service/MonitoredServicePage.constants'
import type { SloHealthIndicatorDTO } from 'services/cv'

export enum RiskValues {
  NO_DATA = 'NO_DATA',
  NO_ANALYSIS = 'NO_ANALYSIS',
  HEALTHY = 'HEALTHY',
  OBSERVE = 'OBSERVE',
  NEED_ATTENTION = 'NEED_ATTENTION',
  UNHEALTHY = 'UNHEALTHY'
}

export enum SLOErrorBudget {
  EXHAUSTED = 'EXHAUSTED'
}

// Need to remove once removed from BE.
type OldRiskTypes = 'LOW' | 'MEDIUM' | 'HIGH'
type RiskTypes = keyof typeof RiskValues | OldRiskTypes

export const getRiskColorValue = (
  riskStatus?: RiskTypes | SloHealthIndicatorDTO['errorBudgetRisk'],
  realCSSColor = true,
  dark = true
): string => {
  const COLOR_NO_DATA = dark ? Color.GREY_400 : Color.GREY_100

  switch (riskStatus) {
    case RiskValues.HEALTHY:
      return realCSSColor ? Utils.getRealCSSColor(Color.GREEN_500) : Color.GREEN_500
    case RiskValues.OBSERVE:
      return realCSSColor ? Utils.getRealCSSColor(Color.YELLOW_800) : Color.YELLOW_800
    case RiskValues.NEED_ATTENTION:
      return realCSSColor ? Utils.getRealCSSColor(Color.ORANGE_600) : Color.ORANGE_600
    case RiskValues.UNHEALTHY:
      return realCSSColor ? Utils.getRealCSSColor(Color.RED_600) : Color.RED_600
    case SLOErrorBudget.EXHAUSTED:
      return realCSSColor ? Utils.getRealCSSColor(Color.RED_800) : Color.RED_800
    default:
      return realCSSColor ? Utils.getRealCSSColor(COLOR_NO_DATA) : COLOR_NO_DATA
  }
}

export function getSecondaryRiskColorValue(
  riskStatus?: RiskTypes | SloHealthIndicatorDTO['errorBudgetRisk'],
  realCSSColor = true
): string {
  switch (riskStatus) {
    case RiskValues.HEALTHY:
      return realCSSColor ? Utils.getRealCSSColor(Color.GREEN_50) : Color.GREEN_50
    case RiskValues.OBSERVE:
      return realCSSColor ? Utils.getRealCSSColor(Color.YELLOW_50) : Color.YELLOW_50
    case RiskValues.NEED_ATTENTION:
      return realCSSColor ? Utils.getRealCSSColor(Color.ORANGE_50) : Color.ORANGE_50
    case RiskValues.UNHEALTHY:
      return realCSSColor ? Utils.getRealCSSColor(Color.RED_50) : Color.RED_50
    case SLOErrorBudget.EXHAUSTED:
      return realCSSColor ? Utils.getRealCSSColor(Color.RED_100) : Color.RED_100
    default:
      return realCSSColor ? Utils.getRealCSSColor(Color.GREY_50) : Color.GREY_50
  }
}

export const getRiskLabelStringId = (
  riskStatus?: RiskTypes | SloHealthIndicatorDTO['errorBudgetRisk']
): keyof StringsMap => {
  switch (riskStatus) {
    case RiskValues.NO_DATA:
      return 'noData'
    case RiskValues.NO_ANALYSIS:
      return 'cv.noAnalysis'
    case RiskValues.HEALTHY:
      return 'cv.monitoredServices.serviceHealth.serviceDependencies.states.healthy'
    case RiskValues.OBSERVE:
      return 'cv.monitoredServices.serviceHealth.serviceDependencies.states.observe'
    case RiskValues.NEED_ATTENTION:
      return 'cv.monitoredServices.serviceHealth.serviceDependencies.states.needsAttention'
    case RiskValues.UNHEALTHY:
      return 'cv.monitoredServices.serviceHealth.serviceDependencies.states.unhealthy'
    case SLOErrorBudget.EXHAUSTED:
      return 'cv.monitoredServices.serviceHealth.serviceDependencies.states.exhausted'
    default:
      return 'na'
  }
}

export function roundNumber(value: number, precision = 2) {
  if (Number.isInteger(precision) && precision >= 0) {
    const factor = 10 ** precision
    return Math.round(value * factor) / factor
  }
}

export function getErrorMessage(errorObj?: any): string | undefined {
  return get(errorObj, 'data.detailedMessage') || get(errorObj, 'data.message') || get(errorObj, 'message')
}

export const getEnvironmentOptions = (
  environmentList: ResponseListEnvironmentResponse | null,
  loading: boolean,
  getString: UseStringsReturn['getString']
): SelectOption[] => {
  if (loading) {
    return [{ label: getString('loading'), value: 'loading' }]
  }
  if (environmentList?.data?.length) {
    const allOption: SelectOption = { label: getString('all'), value: getString('all') }
    const environmentSelectOption: SelectOption[] =
      environmentList?.data?.map((environmentData: EnvironmentResponse) => {
        const { name = '', identifier = '' } = environmentData?.environment || {}
        return {
          label: name,
          value: identifier
        }
      }) || []
    return [allOption, ...environmentSelectOption]
  }
  return []
}

interface GetCVMonitoringServicesSearchParamProps {
  view?: Views
  tab?: MonitoredServiceEnum
}

export const getCVMonitoringServicesSearchParam = ({ view, tab }: GetCVMonitoringServicesSearchParamProps): string => {
  let searchParam = `?`

  if (view === Views.GRID) {
    searchParam = searchParam.concat(`view=${view}&`)
  }
  if (tab) {
    searchParam = searchParam.concat(`tab=${tab}`)
  }

  return searchParam
}

export const prepareFilterInfo = (data?: MultiSelectOption[]): Array<string | number> => {
  return data ? data.map((d: MultiSelectOption) => d.value as string) : []
}

export const isNumeric = (val: string): boolean => {
  return /^-?\d+$/.test(val)
}
