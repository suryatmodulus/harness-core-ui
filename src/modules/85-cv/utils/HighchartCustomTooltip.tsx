/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type {
  Chart as HighchartsChart,
  TooltipFormatterCallbackFunction,
  TooltipFormatterContextObject
} from 'highcharts'
import { useEffect, useState, useRef } from 'react'
import ReactDOM from 'react-dom'

const generateTooltipId = (chartId: number): string => `highcharts-custom-tooltip-${chartId}`

interface Props {
  chart: HighchartsChart | null
  children(formatterContext: TooltipFormatterContextObject): JSX.Element
}

export const HighchartCustomTooltip: React.FC<Props> = ({ chart, children }) => {
  const isInit = useRef(false)
  const [context, setContext] = useState<TooltipFormatterContextObject | null>(null)

  useEffect(() => {
    if (chart?.tooltip && chart.update) {
      const formatter: TooltipFormatterCallbackFunction = function () {
        // Ensures that tooltip DOM container is rendered before React portal is created.
        if (!isInit.current) {
          isInit.current = true

          chart.tooltip.refresh.apply(chart.tooltip, [this.point])
          chart.tooltip.hide(0)
        }

        setContext(this)

        return `<div id="${generateTooltipId(chart.index)}"></div>`
      }

      chart.update({
        tooltip: {
          formatter,
          useHTML: true
        }
      })
    }
  }, [chart])

  const node = chart?.index && document.getElementById(generateTooltipId(chart.index))

  return node && context ? ReactDOM.createPortal(children(context), node) : null
}
