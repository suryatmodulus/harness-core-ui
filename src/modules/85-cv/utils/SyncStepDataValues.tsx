/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { useEffect } from 'react'
import pickBy from 'lodash-es/pickBy'
import merge from 'lodash-es/merge'

interface SyncStepDataValuesProps {
  values: any
  listenToValues: any
  onUpdate: (val: any) => void
}

export default function SyncStepDataValues({ values, listenToValues, onUpdate }: SyncStepDataValuesProps) {
  useEffect(() => {
    const update = pickBy({ ...listenToValues }, value => !!value)
    onUpdate(merge(values, update))
  }, [listenToValues])
  return null
}
