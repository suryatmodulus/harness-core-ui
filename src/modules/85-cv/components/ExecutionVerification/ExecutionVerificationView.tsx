/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useMemo, useState } from 'react'
import { Container, Tabs, Tab, NoDataCard } from '@wings-software/uicore'
import { useStrings } from 'framework/strings'
import { useQueryParams } from '@common/hooks'
import type { ExecutionNode } from 'services/pipeline-ng'
import { DeploymentMetrics } from './components/DeploymentMetrics/DeploymentMetrics'
import { ExecutionVerificationSummary } from './components/ExecutionVerificationSummary/ExecutionVerificationSummary'
import type { DeploymentNodeAnalysisResult } from './components/DeploymentProgressAndNodes/components/DeploymentNodes/DeploymentNodes.constants'
import LogAnalysisContainer from './components/LogAnalysisContainer/LogAnalysisView.container'
import { getActivityId, getDefaultTabId } from './ExecutionVerificationView.utils'
import css from './ExecutionVerificationView.module.scss'

interface ExecutionVerificationViewProps {
  step: ExecutionNode
}

export function ExecutionVerificationView(props: ExecutionVerificationViewProps): JSX.Element {
  const { step } = props
  const { getString } = useStrings()
  const [selectedNode, setSelectedNode] = useState<DeploymentNodeAnalysisResult | undefined>()
  const activityId = useMemo(() => getActivityId(step), [step])
  const { type } = useQueryParams<{ type?: string }>()
  const defaultTabId = useMemo(() => getDefaultTabId(getString, type), [type])

  const content = activityId ? (
    <Tabs id="AnalysisTypeTabs" defaultSelectedTabId={defaultTabId}>
      <Tab
        id={getString('pipeline.verification.analysisTab.metrics')}
        title={getString('pipeline.verification.analysisTab.metrics')}
        panel={
          <DeploymentMetrics step={step} selectedNode={selectedNode} activityId={activityId as unknown as string} />
        }
      />
      <Tab
        id={getString('pipeline.verification.analysisTab.logs')}
        title={getString('pipeline.verification.analysisTab.logs')}
        panel={<LogAnalysisContainer step={step} hostName={selectedNode?.hostName} />}
      />
    </Tabs>
  ) : (
    <Container className={css.noAnalysis}>
      <NoDataCard message={getString('pipeline.verification.logs.noAnalysis')} icon="warning-sign" />
    </Container>
  )
  return (
    <Container className={css.main}>
      <ExecutionVerificationSummary
        displayAnalysisCount={false}
        step={step}
        className={css.executionSummary}
        onSelectNode={setSelectedNode}
      />
      {content}
    </Container>
  )
}
