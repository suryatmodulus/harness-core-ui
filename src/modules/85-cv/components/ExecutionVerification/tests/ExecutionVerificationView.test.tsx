/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, waitFor } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import type { ExecutionNode } from 'services/pipeline-ng'
import { ExecutionVerificationView } from '../ExecutionVerificationView'
import { getActivityId, getDefaultTabId } from '../ExecutionVerificationView.utils'

jest.mock('../components/DeploymentMetrics/DeploymentMetrics', () => ({
  ...(jest.requireActual('../components/DeploymentMetrics/DeploymentMetrics') as any),
  DeploymentMetrics: () => <div className="deploymentMetrics" />
}))

jest.mock('../components/ExecutionVerificationSummary/ExecutionVerificationSummary', () => ({
  ExecutionVerificationSummary: () => <div className="summary" />
}))

jest.mock('../components/LogAnalysisContainer/LogAnalysisView.container', () => ({
  __esModule: true,
  default: () => <div className="LogAnalysisContainer" />
}))

jest.mock('highcharts-react-official', () => () => <></>)

describe('Unit tests for ExecutionVerificationView unit tests', () => {
  test('Ensure tabs are rendered', async () => {
    const { container, getByText } = render(
      <TestWrapper>
        <ExecutionVerificationView step={{ progressData: { activityId: '1234_activityId' as any } }} />
      </TestWrapper>
    )
    await waitFor(() => expect(container.querySelector('[class*="bp3-tabs"]')).not.toBeNull())
    expect(getByText('pipeline.verification.analysisTab.logs')).not.toBeNull()
    expect(getByText('pipeline.verification.analysisTab.metrics')).not.toBeNull()
    expect(container).toMatchSnapshot()
  })
  test('Ensure no analysis state is rendered when activity id does not exist', async () => {
    const { container } = render(
      <TestWrapper>
        <ExecutionVerificationView step={{}} />
      </TestWrapper>
    )

    await waitFor(() => expect(container.querySelector('[class*="noAnalysis"]')))
    expect(container.querySelector('[class*="bp3-tabs"]')).toBeNull()
    expect(container).toMatchSnapshot()
  })

  test('Ensure correct activityId is returned when getActivityId utils method is called with activityId being present in progressdata and not in outputdata', async () => {
    const step = {
      progressData: {
        activityId: 'activityId-from-step-progressData'
      }
    }
    expect(getActivityId(step as unknown as ExecutionNode)).toEqual('activityId-from-step-progressData')
  })

  test('Ensure correct activityId is returned when getActivityId utils method is called with activityId is not present in progressdata but present in output data', async () => {
    const step = {
      outcomes: {
        output: {
          activityId: 'activityId-from-step-outputdata'
        }
      }
    }
    expect(getActivityId(step as unknown as ExecutionNode)).toEqual('activityId-from-step-outputdata')
  })

  test('Ensure correct tabId is returned  ', () => {
    jest.mock('framework/strings', () => ({
      useStrings: () => ({
        getString: (val: string) => val
      })
    }))
    expect(getDefaultTabId((item: string) => item, undefined)).toEqual('pipeline.verification.analysisTab.metrics')
    expect(getDefaultTabId((item: string) => item, 'pipeline.verification.analysisTab.logs')).toEqual(
      'pipeline.verification.analysisTab.logs'
    )
    expect(getDefaultTabId((item: string) => item, 'pipeline.verification.analysisTab.metrics')).toEqual(
      'pipeline.verification.analysisTab.metrics'
    )
  })

  test('Ensure correct tabs are rendered via queryParams', async () => {
    const LogsContainer = render(
      <TestWrapper queryParams={{ type: 'pipeline.verification.analysisTab.logs' }}>
        <ExecutionVerificationView step={{ progressData: { activityId: '1234_activityId' as any } }} />
      </TestWrapper>
    )
    expect(LogsContainer.container.querySelector('.LogAnalysisContainer')).toBeInTheDocument()
    expect(LogsContainer.container.querySelector('.deploymentMetrics')).not.toBeInTheDocument()
    expect(LogsContainer.container).toMatchSnapshot()

    const MetricsContainer = render(
      <TestWrapper queryParams={{ type: 'pipeline.verification.analysisTab.metrics' }}>
        <ExecutionVerificationView step={{ progressData: { activityId: '1234_activityId' as any } }} />
      </TestWrapper>
    )
    expect(MetricsContainer.container.querySelector('.LogAnalysisContainer')).not.toBeInTheDocument()
    expect(MetricsContainer.container.querySelector('.deploymentMetrics')).toBeInTheDocument()
    expect(MetricsContainer.container).toMatchSnapshot()
  })
})
