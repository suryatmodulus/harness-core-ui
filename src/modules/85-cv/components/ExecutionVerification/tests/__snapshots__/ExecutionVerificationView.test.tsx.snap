// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for ExecutionVerificationView unit tests Ensure correct tabs are rendered via queryParams 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main main"
  >
    <div
      class="summary"
    />
    <div
      class="bp3-tabs Tabs--main Tabs--horizontal"
    >
      <div
        class="bp3-tab-list"
        role="tablist"
      >
        <div
          class="bp3-tab-indicator-wrapper"
          style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
        >
          <div
            class="bp3-tab-indicator"
          />
        </div>
        <div
          aria-controls="bp3-tab-panel_AnalysisTypeTabs_pipeline.verification.analysisTab.metrics"
          aria-disabled="false"
          aria-expanded="false"
          aria-selected="false"
          class="bp3-tab"
          data-tab-id="pipeline.verification.analysisTab.metrics"
          id="bp3-tab-title_AnalysisTypeTabs_pipeline.verification.analysisTab.metrics"
          role="tab"
          tabindex="0"
        >
          pipeline.verification.analysisTab.metrics
        </div>
        <div
          aria-controls="bp3-tab-panel_AnalysisTypeTabs_pipeline.verification.analysisTab.logs"
          aria-disabled="false"
          aria-expanded="true"
          aria-selected="true"
          class="bp3-tab"
          data-tab-id="pipeline.verification.analysisTab.logs"
          id="bp3-tab-title_AnalysisTypeTabs_pipeline.verification.analysisTab.logs"
          role="tab"
          tabindex="0"
        >
          pipeline.verification.analysisTab.logs
        </div>
      </div>
      <div
        aria-hidden="false"
        aria-labelledby="bp3-tab-title_AnalysisTypeTabs_pipeline.verification.analysisTab.logs"
        class="bp3-tab-panel"
        id="bp3-tab-panel_AnalysisTypeTabs_pipeline.verification.analysisTab.logs"
        role="tabpanel"
      >
        <div
          class="LogAnalysisContainer"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for ExecutionVerificationView unit tests Ensure correct tabs are rendered via queryParams 2`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main main"
  >
    <div
      class="summary"
    />
    <div
      class="bp3-tabs Tabs--main Tabs--horizontal"
    >
      <div
        class="bp3-tab-list"
        role="tablist"
      >
        <div
          class="bp3-tab-indicator-wrapper"
          style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
        >
          <div
            class="bp3-tab-indicator"
          />
        </div>
        <div
          aria-controls="bp3-tab-panel_AnalysisTypeTabs_pipeline.verification.analysisTab.metrics"
          aria-disabled="false"
          aria-expanded="true"
          aria-selected="true"
          class="bp3-tab"
          data-tab-id="pipeline.verification.analysisTab.metrics"
          id="bp3-tab-title_AnalysisTypeTabs_pipeline.verification.analysisTab.metrics"
          role="tab"
          tabindex="0"
        >
          pipeline.verification.analysisTab.metrics
        </div>
        <div
          aria-controls="bp3-tab-panel_AnalysisTypeTabs_pipeline.verification.analysisTab.logs"
          aria-disabled="false"
          aria-expanded="false"
          aria-selected="false"
          class="bp3-tab"
          data-tab-id="pipeline.verification.analysisTab.logs"
          id="bp3-tab-title_AnalysisTypeTabs_pipeline.verification.analysisTab.logs"
          role="tab"
          tabindex="0"
        >
          pipeline.verification.analysisTab.logs
        </div>
      </div>
      <div
        aria-hidden="false"
        aria-labelledby="bp3-tab-title_AnalysisTypeTabs_pipeline.verification.analysisTab.metrics"
        class="bp3-tab-panel"
        id="bp3-tab-panel_AnalysisTypeTabs_pipeline.verification.analysisTab.metrics"
        role="tabpanel"
      >
        <div
          class="deploymentMetrics"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for ExecutionVerificationView unit tests Ensure no analysis state is rendered when activity id does not exist 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main main"
  >
    <div
      class="summary"
    />
    <div
      class="StyledProps--font StyledProps--main noAnalysis"
    >
      <div
        class="StyledProps--font StyledProps--main NoDataCard--noDataCard StyledProps--flex StyledProps--flex-align-center-center"
      >
        <div
          class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
          style="align-items: center; margin-top: -48px; width: 470px;"
        >
          <span
            class="bp3-icon bp3-icon-warning-sign StyledProps--main StyledProps--color StyledProps--color-grey600"
            icon="warning-sign"
          >
            <svg
              data-icon="warning-sign"
              height="48"
              viewBox="0 0 20 20"
              width="48"
            >
              <desc>
                warning-sign
              </desc>
              <path
                d="M19.86 17.52l.01-.01-9-16-.01.01C10.69 1.21 10.37 1 10 1s-.69.21-.86.52l-.01-.01-9 16 .01.01c-.08.14-.14.3-.14.48 0 .55.45 1 1 1h18c.55 0 1-.45 1-1 0-.18-.06-.34-.14-.48zM11 17H9v-2h2v2zm0-3H9V6h2v8z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <p
            class="StyledProps--font StyledProps--main NoDataCard--message StyledProps--font-align-center StyledProps--color StyledProps--color-grey600"
          >
            pipeline.verification.logs.noAnalysis
          </p>
          <div
            class="NoDataCard--buttonContainer"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for ExecutionVerificationView unit tests Ensure tabs are rendered 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main main"
  >
    <div
      class="summary"
    />
    <div
      class="bp3-tabs Tabs--main Tabs--horizontal"
    >
      <div
        class="bp3-tab-list"
        role="tablist"
      >
        <div
          class="bp3-tab-indicator-wrapper"
          style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
        >
          <div
            class="bp3-tab-indicator"
          />
        </div>
        <div
          aria-controls="bp3-tab-panel_AnalysisTypeTabs_pipeline.verification.analysisTab.metrics"
          aria-disabled="false"
          aria-expanded="true"
          aria-selected="true"
          class="bp3-tab"
          data-tab-id="pipeline.verification.analysisTab.metrics"
          id="bp3-tab-title_AnalysisTypeTabs_pipeline.verification.analysisTab.metrics"
          role="tab"
          tabindex="0"
        >
          pipeline.verification.analysisTab.metrics
        </div>
        <div
          aria-controls="bp3-tab-panel_AnalysisTypeTabs_pipeline.verification.analysisTab.logs"
          aria-disabled="false"
          aria-expanded="false"
          aria-selected="false"
          class="bp3-tab"
          data-tab-id="pipeline.verification.analysisTab.logs"
          id="bp3-tab-title_AnalysisTypeTabs_pipeline.verification.analysisTab.logs"
          role="tab"
          tabindex="0"
        >
          pipeline.verification.analysisTab.logs
        </div>
      </div>
      <div
        aria-hidden="false"
        aria-labelledby="bp3-tab-title_AnalysisTypeTabs_pipeline.verification.analysisTab.metrics"
        class="bp3-tab-panel"
        id="bp3-tab-panel_AnalysisTypeTabs_pipeline.verification.analysisTab.metrics"
        role="tabpanel"
      >
        <div
          class="deploymentMetrics"
        />
      </div>
    </div>
  </div>
</div>
`;
